src/main/java/vimification/taskui/TaskDetailPanel.java

```diff
@@ -0,0 +1,38 @@
+package vimification.taskui;
+
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.VBox;
+import vimification.model.task.Task;
+import javafx.fxml.FXML;
+
+/**
+ * Panel containing task details, such as description, the type of the task, etc.
+ */
+public class TaskDetailPanel extends UiPart<VBox> {
+    private static final String FXML = "TaskDetailPanel.fxml";
+
+    private Task task;
+
+    @FXML
+    private ListView<Task> taskListView;
+    @FXML
+    private Label descriptionText;
+    @FXML
+    private Label typeText;
+
+    /**
+     * Creates a {@code TaskDetailPanel}.
+     */
+    public TaskDetailPanel(Task task) {
+        super(FXML);
+        this.task = task;
+        setup();
+    }
+
+    public void setup() {
```

Comment:
Rename to `initialize` and add `@FMXL` - make it consistent.

Reply:
This is intentional, we cannot use @FXML and initialize because it's a special marker to signalize an event of the initialize life cycle method of JavaFX that is triggered before loading and instantiating the component instance.

What I want is instead, setting up the component after the component instance has been instantiated.

Reply:
Source:
https://stackoverflow.com/questions/34785417/javafx-fxml-controller-constructor-vs-initialize-method